SofascoreAcademy/SofascoreAcademy/Views/MainViewController.swift

```diff
@@ -71,46 +116,157 @@ extension MainViewController: BaseViewProtocol {
             $0.leading.trailing.equalToSuperview()
         }
         
-        containerView.snp.makeConstraints() {
+        datePickerView.snp.makeConstraints(){
             $0.top.equalTo(customTabBar.snp.bottom)
+            $0.leading.trailing.equalToSuperview()
+            $0.height.equalTo(48)
+        }
+        
+        datesMatchesDivider.snp.makeConstraints() {
+            $0.top.equalTo(datePickerView.snp.bottom)
+            $0.leading.trailing.equalToSuperview()
+        }
+        
+        containerView.snp.makeConstraints() {
+            $0.top.equalTo(datesMatchesDivider.snp.bottom)
             $0.leading.trailing.bottom.equalToSuperview()
         }
     }
 }
 
+// MARK: AppHeaderDelegate
+extension MainViewController: AppHeaderDelegate {
+    
+    func reactToSetingsTap() {
+        let settingsViewController = SettingsViewController()
+        settingsViewController.modalPresentationStyle = .fullScreen
+        settingsViewController.title = "Settings"
+        present(settingsViewController, animated: true)
+    }
+}
+
+// MARK: MatchTapDelegate
+extension MainViewController: MatchTapDelegate {
+    
+    func displayMatchInfoOnTap(selectedMatch: Event) {
+        navigationController?.pushViewController(MatchDataViewController(matchData: selectedMatch), animated: true)
+    }
+}
+
+//MARK: DatePickDelegate
+extension MainViewController: DatePickDelegate {
+    
+    func displayEventsForSelectedDate(selectedDate: String) {
+        self.selectedDate = selectedDate
+        currentChild.remove()
+        startLoading()
+        
+        reloadData(selectedDate: selectedDate, selectedSport: currentSportSlug)
```

Comment:
Malo me muci ovaj naming reloadData i moze biti misleading. Prvi put kad se load-a, to zapravo nije reload nego load. Mozes napraviti load i reload fju i onda da reload zove load